pgxn/neon/file_cache.c

```diff
@@ -66,6 +68,66 @@
 #define MAX_MONITOR_INTERVAL_USEC 1000000 /* 1 second */
 #define MAX_DISK_WRITE_RATE       1000 /* MB/sec */
 
+/* Hyper log-log implementation for counting size of working set */
+
+#define HASH_BITS 25
+#define N_HASHES (1 << (32 - HASH_BITS))
+
+static void
+calculate_zero_bits(uint32 h, uint8* max_zero_bits)
+{
+    int j = h >> HASH_BITS;
+    int zero_bits = 1;
+    while ((h & 1) == 0 && zero_bits <= HASH_BITS)  {
+        h >>= 1;
+        zero_bits += 1;
+    }
+    if (max_zero_bits[j] < zero_bits) {
+        max_zero_bits[j] = zero_bits;
+    }
+}
+
+static void
+calculate_hash_functions(void const* val, int size, uint8* max_zero_bits)
+{
+    uint32 h = hash_bytes((char const*)val, size);
+    calculate_zero_bits(h, max_zero_bits);
+}
+
+static uint32
+approximate_distinct_count(uint8* max_zero_bits)
+{
+    const int m = N_HASHES;
+    const double alpha_m = 0.7213 / (1 + 1.079 / (double)m);
+    const double pow_2_32 = 0xffffffff;
```

Comment:
It's not true. `1 << 32` is `pow(2,32)`. Here you have `2 ** 32 - 1`. I assume that it was the idea, but the name misleads